src/Raven.Server/ServerWide/DiskIOTester.cs

```diff
@@ -0,0 +1,904 @@
+        {
+            if (secondTimer == null)
+                return;
+
+            secondTimer.Dispose();
+            secondTimer = null;
+        }
+    }
+
+    public class GenericDiskPerformanceTester : AbstractDiskPerformanceTester<GenericPerformanceTestRequest>
+    {
+
+        private readonly List<Thread> threads;
+        private readonly List<Random> perThreadRandom;
+
+        public GenericDiskPerformanceTester(GenericPerformanceTestRequest testRequest, CancellationToken taskKillToken = new CancellationToken()) : base(testRequest, taskKillToken)
+        {
+            IOExtensions.CreateDirectoryIfNotExists(testRequest.Path);
+            filePath = Path.Combine(testRequest.Path, TemporaryFileName);
+            threads = new List<Thread>(testRequest.ThreadCount);
+            perThreadRandom = Enumerable.Range(1, testRequest.ThreadCount)
+                .Select(i => testRequest.RandomSeed.HasValue ? new Random(testRequest.RandomSeed.Value) : new Random()).ToList();
+
+            if (testRequest.Sequential && testRequest.OperationType == OperationType.Mix)
+            {
+                onInfo("Sequential test with mixed read/write mode is not supported. Changing to random access");
+                testRequest.Sequential = false;
+            }
+        }
+
+        public override IOperationResult TestDiskIO(Action<IOperationProgress> onProgress)
+        {
+            onInfo = str => onProgress(new IOTestOperationProgress
+            {
+                Message = str
+            });
+
+            PrepareTestFile(filePath);            
+            try
+            {
+                using (TestTimeMeasure())
+                {
+                    StartWorkers();
+                    onInfo("Waiting for test to complete");
+                    threads.ForEach(t => t.Join());
+
+                    taskKillToken.ThrowIfCancellationRequested();
+                }
+            }
+            catch (AggregateException e)
+            {
+                var result = GetResultFromDataStorage("Failed the test with errors.");
+                result.Errors = e.InnerExceptions.Select(x => x.ToString()).ToArray();
+
+                return result;
+            }
+            catch (Exception e)
+            {
+                var result = GetResultFromDataStorage("Failed the test with errors.");
+                result.Errors = new[] {e.ToString()};
+
+                return result;
+            }
+
+            return GetResultFromDataStorage("Completed the test successfully.");
+        }
+
+        private void StartWorkers()
+        {
+            var stripeSize = testRequest.FileSize / testRequest.ThreadCount;
+            for (var i = 0; i < testRequest.ThreadCount; i++)
+            {
+                var threadNumber = i;
+                threads.Add(new Thread(() => MeasurePerformance(perThreadRandom[threadNumber], threadNumber * stripeSize, (threadNumber + 1) * stripeSize,linkedCts.Token)));
```

Comment:
Add thread names, of better yet, just schedule this using PoolOfThreads, I don't think we need dedicated threads here now